tests/test_unit/test_vector.py

```diff
@@ -121,3 +121,54 @@ def test_pol2cart(self, ds, expected_exception, request):
         with expected_exception:
             result = vector.pol2cart(ds.pol)
             xr.testing.assert_allclose(result, ds.cart)
+
+    @pytest.mark.parametrize(
+        "ds, expected_exception",
+        [
+            ("cart_pol_dataset", does_not_raise()),
+            ("cart_pol_dataset_with_nan", does_not_raise()),
+            ("cart_pol_dataset_missing_cart_dim", pytest.raises(ValueError)),
+            (
+                "cart_pol_dataset_missing_cart_coords",
+                pytest.raises(ValueError),
+            ),
+        ],
+    )
+    def test_magnitude(self, ds, expected_exception, request):
+        """Test vector magnitude with known values."""
+        ds = request.getfixturevalue(ds)
+        with expected_exception:
+            result = vector.magnitude(ds.cart)
+            expected = np.sqrt(
+                ds.cart.sel(space="x") ** 2 + ds.cart.sel(space="y") ** 2
+            )
+            xr.testing.assert_allclose(result, expected)
+            # result should be the same from Cartesian and polar coordinates
+            xr.testing.assert_allclose(result, vector.magnitude(ds.pol))
+            # The result should only contain the time dimension.
+            assert result.dims == ("time",)
+
+    @pytest.mark.parametrize(
+        "ds, expected_exception",
+        [
+            ("cart_pol_dataset", does_not_raise()),
+            ("cart_pol_dataset_with_nan", does_not_raise()),
+            ("cart_pol_dataset_missing_cart_dim", pytest.raises(ValueError)),
+        ],
+    )
+    def test_normalize(self, ds, expected_exception, request):
+        """Test data normalization (division by magnitude)."""
+        ds = request.getfixturevalue(ds)
+        with expected_exception:
+            normalized = vector.normalize(ds.cart)
+            # the normalized data should have the same dimensions as the input
+            assert normalized.dims == ds.cart.dims
+            # the first time point is NaN because the input vector is [0, 0]
+            # (zero-division during normalization).
+            assert normalized.sel(time=0).isnull().all()
+            # the magnitude of the normalized vector should be 1 for all
+            # time points except for the expected NaNs.
+            normalized_mag = vector.magnitude(normalized).values
```

Comment:
I've now changed this test quite a bit, taking into account all your comments:

```python
   @pytest.mark.parametrize(
        "ds, expected_exception",
        [
            ("cart_pol_dataset", does_not_raise()),
            ("cart_pol_dataset_with_nan", does_not_raise()),
            ("cart_pol_dataset_missing_cart_dim", pytest.raises(ValueError)),
        ],
    )
    def test_convert_to_unit(self, ds, expected_exception, request):
        """Test conversion to unit vectors (normalisation)."""
        ds = request.getfixturevalue(ds)
        with expected_exception:
            # normalise both the Cartesian and the polar data to unit vectors
            unit_cart = vector.convert_to_unit(ds.cart)
            unit_pol = vector.convert_to_unit(ds.pol)
            # they should yield the same result, just in different coordinates
            xr.testing.assert_allclose(unit_cart, vector.pol2cart(unit_pol))
            xr.testing.assert_allclose(unit_pol, vector.cart2pol(unit_cart))

            # since we established that polar vs Cartesian unit vectors are
            # equivalent, it's enough to do other assertions on either one

            # the normalised data should have the same dimensions as the input
            assert unit_cart.dims == ds.cart.dims

            # unit vector should be NaN if the input vector was null or NaN
            is_null_vec = (ds.cart == 0).all("space")  # null vec: x=0, y=0
            is_nan_vec = ds.cart.isnull().any("space")  # any NaN in x or y
            expected_nan_idxs = is_null_vec | is_nan_vec
            assert unit_cart.where(expected_nan_idxs).isnull().all()

            # For non-NaN unit vectors in polar coordinates, the rho values
            # should be 1 and the phi values should be the same as the input
            expected_unit_pol = ds.pol.copy()
            expected_unit_pol.loc[{"space_pol": "rho"}] = 1
            expected_unit_pol = expected_unit_pol.where(~expected_nan_idxs)
            xr.testing.assert_allclose(unit_pol, expected_unit_pol)
```

I think I now cover what you pointed out.